### PR TITLE
1.0 migration blog post

### PR DESCRIPTION
**Pull request by orta**
Alright, should cover a lot of the whys and provide some good examples of before/afters for common looking podfiles

### CODE REVIEW

File: _posts/2016-01-29-CocoaPods-1.0.markdown [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+author: orta
+categories: cocoapods release
+---
+
+#### Major 1.0 changes TL:DR
+​
+
+- You now have to define targets, previously there was an available implicit target. Basically all `pod`s have to be inside a `target`.
+- A `target` has to represent an Xcode target.
+- When you want to represent a collection of pods that go to multiple targets, use an `abstract_target` then add targets inside that. Pod dependencies are inherited by nested targets.
```
Comment (neonichu) [inline _posts/2016-01-29-CocoaPods-1.0.markdown]: Would mention here that the former implicit target is now an abstract one.

Comment (orta) [inline _posts/2016-01-29-CocoaPods-1.0.markdown]: does that still work?

Comment (orta) [inline _posts/2016-01-29-CocoaPods-1.0.markdown]: e.g.

```ruby
abstract_target 'Shows' do
  pod 'ShowsKit'
  pod 'Fabric'

  # Has it's own copy of ShowsKit + ShowWebAuth
  target 'ShowsiOS' do
    pod 'ShowWebAuth'
  end

  # Has it's own copy of ShowsKit + ShowTVAuth
  target 'ShowsTV' do
    pod 'ShowTVAuth'
  end
end
```
==

``` ruby
pod 'ShowsKit'
pod 'Fabric'

# Has it's own copy of ShowsKit + ShowWebAuth
target 'ShowsiOS' do
  pod 'ShowWebAuth'
end

# Has it's own copy of ShowsKit + ShowTVAuth
target 'ShowsTV' do
  pod 'ShowTVAuth'
end

```

Comment (neonichu) [inline _posts/2016-01-29-CocoaPods-1.0.markdown]: Yep, that should work.

File: _posts/2016-01-29-CocoaPods-1.0.markdown [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+- Targets that want to know about the pods for a target, but don't need linking (for example, Test targets) can define that they inherit pods via their search paths (`inherit! :search_paths`).
+- `:exclusive => true` and `link_with` have been removed in favour of the above.
+- `xcodeproj` is renamed `project` to make the DSL vocabulary consistent.
+- We've removed the ability to declare `:head` on a `pod` -- you can either use the `:podspec` option or point directly to the upstream repo (via `git`, `:svn`, etc.)
+- You can now declare installation options inside the `Podfile` - via `install!`. (i.e. `install! 'cocoapods', :deterministic_uuids => false`)
```
Comment (neonichu) [inline _posts/2016-01-29-CocoaPods-1.0.markdown]: We should mention that `—no-integrate` is such an option now `integrate_targets => false`

Comment (orta) [inline _posts/2016-01-29-CocoaPods-1.0.markdown]: I've mentioned it in the body, I need to see where they're expose in the docs really and provide a link to the guides

Comment (orta) [inline _posts/2016-01-29-CocoaPods-1.0.markdown]: https://github.com/CocoaPods/CocoaPods/blob/master/lib/cocoapods/installer/installation_options.rb#L104-L108

Hrm these might not be exposed anywhere

Comment (neonichu) [inline _posts/2016-01-29-CocoaPods-1.0.markdown]: I saw, but IMO belongs in the tl;dr :)

Comment (neonichu) [inline _posts/2016-01-29-CocoaPods-1.0.markdown]: The live guides mention it at least <https://guides.cocoapods.org/syntax/podfile.html#install_bang>

Comment (orta) [inline _posts/2016-01-29-CocoaPods-1.0.markdown]: Cool, added an issue about adding the details to that: https://github.com/CocoaPods/CocoaPods/issues/4842
I can add that up there.

File: _posts/2016-01-29-CocoaPods-1.0.markdown [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+layout: post
+title:  "CocoaPods 1.0 Migration Guide"
+author: orta
+categories: cocoapods release
+---
+
+#### Major 1.0 changes TL:DR
+​
+
+- You now have to define targets, previously there was an available implicit target. Basically all `pod`s have to be inside a `target`.
```
Comment (segiddins) [inline _posts/2016-01-29-CocoaPods-1.0.markdown]: not true. pods at the root level are still OK, they'll just be inherited

File: _posts/2016-01-29-CocoaPods-1.0.markdown [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+### Off with `pod`s `:head`
+
+Adding `:head` to a `pod` nearly always worked, except when it didn't. It is documented as this:
+
+> Finally, instead of a version, you can specify the :head flag. This will use the pod’s latest version spec version, but force the download of the ‘bleeding edge’ version. Use this with caution, as the spec might not be compatible with the source material anymore.
+
+It was a tool that was really useful back when CocoaPods was small and it was hard to persuade people it was worth support any dependency manager for their projects. That time has long passed and now the flexibility that this attribute originally offered has fallen onto the side of "not worth keeping around."It would introduce extremely hard to debug issues for developers expecting it to be "use the latest version of their code repo." You can either use the `:podspec` option or point directly to the upstream repo (via `git`, `:svn`, etc.)
```
Comment (segiddins) [inline _posts/2016-01-29-CocoaPods-1.0.markdown]: missing space after `.`

File: _posts/2016-01-29-CocoaPods-1.0.markdown [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+### Building for stability
+
+A lot of the work for 1.0 has been around trying to simplify CocoaPods in  order to provide long-lasting infrastructure. This is especially true given that the foundations that CocoaPods builds upon often shifts [every June](https://en.wikipedia.org/wiki/Apple_Worldwide_Developers_Conference) and the project is ran by a handful of people in their spare time.
+
+We also understand that running CocoaPods is about dealing with humans, our default CocoaPods Specs repo for example has had  a policy of being read-only since we introduced [CocoaPods Trunk](http://blog.cocoapods.org/CocoaPods-Trunk/). In reality though, we've had to support editing and deleting Pods by hand on a relatively regular basis with the CocoaPods dev team being the ones having to make those decisions. With 1.0.0 we've built the infrastructure to support deleting and deprecating a CocoaPod and a CocoaPod version. We don't recommend it. It's now there, and it's up to library owners to make the decisions around whether it's fine for their users to remove dependencies.
```
Comment (segiddins) [inline _posts/2016-01-29-CocoaPods-1.0.markdown]: `on our`

**Comment by orta** [comment]
thanks all, feedback addressed

File: _posts/2016-01-29-CocoaPods-1.0.markdown [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+It was a tool that was really useful back when CocoaPods was small and it was hard to persuade people it was worth support any dependency manager for their projects. That time has long passed and now the flexibility that this attribute originally offered has fallen onto the side of "not worth keeping around."It would introduce extremely hard to debug issues for developers expecting it to be "use the latest version of their code repo." You can either use the `:podspec` option or point directly to the upstream repo ( via `git`, `:svn`, etc. )
+
+
+### Options to `install!`
+
+In the past when we have had to support configuration options for CocoaPods we have used environment variables. These act like NSUserDefaults in that someone who knows the key can make a change, but otherwise they remain [quite hidden](https://github.com/ChatSecure/ChatSecure-iOS/commit/d766400c2d8e935ab49daf5ef5f9bf28c760e484). Though, sometimes they could be command line APIs too like `--no-integrate`. These were a conscious choices that eventually added up to being inconsistent over time, and we've opted to move away from that to centralizing the information in the Podfile.
```
Comment (neilkimmett) [inline _posts/2016-01-29-CocoaPods-1.0.markdown]: These were ~~a~~ conscious choices

File: _posts/2016-01-29-CocoaPods-1.0.markdown [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+
+This means that every developer on your team would have the same output on `pod install` regardless of their environment. Whether in [CocoaPods.app](https://cocoapods.org/app) or in the command line. We've left the DSL attribute `install!` quite open so that [future features](https://github.com/CocoaPods/CocoaPods/issues/2729) can be built with this attribute.
+
+### Building for stability
+
+A lot of the work for 1.0 has been around trying to simplify CocoaPods in order to provide long-lasting infrastructure. This is especially true given that the foundations that CocoaPods builds upon often shifts [every June](https://en.wikipedia.org/wiki/Apple_Worldwide_Developers_Conference) and the project is ran by a handful of people in their spare time.
```
Comment (neilkimmett) [inline _posts/2016-01-29-CocoaPods-1.0.markdown]: project is ~~ran~~ run by a handful of people

Comment (orta) [inline _posts/2016-01-29-CocoaPods-1.0.markdown]: dave-gibson-and-charlie-talbot-battle-of-britain-north-vs-south_22140

but yeah, my N.Irish girlfriend would tell me to change that, so I should do it :dancers:

File: _posts/2016-01-29-CocoaPods-1.0.markdown (hunk 6 above)
Comment (orta) [inline _posts/2016-01-29-CocoaPods-1.0.markdown]: :+1: thanks sir

File: _posts/2016-01-29-CocoaPods-1.0.markdown [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+
+### Off with `pod`s `:head`
+
+Adding `:head` to a `pod` nearly always worked, except when it didn't. It is documented as this:
+
+> Finally, instead of a version, you can specify the :head flag. This will use the pod’s latest version spec version, but force the download of the ‘bleeding edge’ version. Use this with caution, as the spec might not be compatible with the source material anymore.
+
+It was a tool that was really useful back when CocoaPods was small and it was hard to persuade people it was worth support any dependency manager for their projects. That time has long passed and now the flexibility that this attribute originally offered has fallen onto the side of "not worth keeping around."It would introduce extremely hard to debug issues for developers expecting it to be "use the latest version of their code repo." You can either use the `:podspec` option or point directly to the upstream repo ( via `git`, `:svn`, etc. )
```
Comment (AliSoftware) [inline _posts/2016-01-29-CocoaPods-1.0.markdown]: `"not worth keeping around". ` (switch quote & full stop + space)

Comment (orta) [inline _posts/2016-01-29-CocoaPods-1.0.markdown]: in british English we keep it inside the quotes, but I'll definitely add a space - thanks!

Comment (neonichu) [inline _posts/2016-01-29-CocoaPods-1.0.markdown]: :gb: is the worst

Comment (orta) [inline _posts/2016-01-29-CocoaPods-1.0.markdown]: :de: is the wurst

**Comment by orta** [comment]
Alright, this is happening with B3 out now

**Comment by neonichu** [comment]
:clap:

**Pull request merged**